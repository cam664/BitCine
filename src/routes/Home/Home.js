import React from 'react';
import StarwarsList from '../../features/StarwarsList';
import Table from '../../common/Table';

/** @jsx jsx */
import { css, jsx } from '@emotion/core';
import styles from './HomeStyles';

export const Home = () => {
  return (
    <main css={css`${styles.main}`}>
      <StarwarsList />
    </main>
  );
}


