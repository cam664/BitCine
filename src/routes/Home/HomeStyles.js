export default {
  main: `
    padding: 120px 2.5% 90px;
    max-width: 1680px;
    margin: 0 auto;
  `
}