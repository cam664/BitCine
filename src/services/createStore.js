import { createStore, combineReducers, applyMiddleware } from "redux";
import thunkMiddleware from "redux-thunk";

import { StarwarsListReducer } from '../features/StarwarsList';

const reducer = combineReducers({
  StarwarsList: StarwarsListReducer,
});

export default (initialState) => 
  createStore(
    reducer, 
    initialState, 
    applyMiddleware(thunkMiddleware)
  );
