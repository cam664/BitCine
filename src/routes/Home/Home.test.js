import React from 'react';
import renderer from 'react-test-renderer';

import Home from './Home';
import { Provider as ReduxProvider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

import createStore from '../../services/createStore';

const store = createStore(window.__PRELOADED_STATE__);

describe('Home Route', () => {
  it('renders a snapshot', () => {
    const component = renderer.create(
      <ReduxProvider store={store}>
        <Router>
          <Home />
        </Router>
      </ReduxProvider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});