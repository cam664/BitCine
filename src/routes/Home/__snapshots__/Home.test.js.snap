// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Route renders a snapshot 1`] = `
<main>
  <section
    className="css-1rt94nh-Table"
  >
    <table>
      <thead>
        <tr>
          <th
            style={
              Object {
                "width": "40%",
              }
            }
          >
            <div>
              Name
            </div>
          </th>
          <th
            style={
              Object {
                "width": "20%",
              }
            }
          >
            <div>
              Birth Year
            </div>
          </th>
          <th
            style={
              Object {
                "width": "20%",
              }
            }
          >
            <div>
              Height
            </div>
          </th>
          <th
            style={
              Object {
                "width": "20%",
              }
            }
          >
            <div>
              Mass
            </div>
          </th>
        </tr>
      </thead>
    </table>
    <div
      className="css-1mzwwa2-Table"
      onScroll={[Function]}
    >
      <table>
        <tbody />
      </table>
    </div>
  </section>
</main>
`;
